src/virtual_file.rs

```diff
@@ -0,0 +1,31 @@
+use crate::common::*;
+use cradle::prelude::*;
+use tempfile::TempDir;
+
+pub(crate) async fn serve(config: Config, file_name: &str) -> Option<Response<Body>> {
+  match config.files.get(file_name) {
+    None => None,
+    Some(virtual_file) => {
+      let VirtualFile::script { source } = virtual_file;
+      let source = if dbg!(source).starts_with("#!") {
+        source.to_string()
+      } else {
+        format!("#!/usr/bin/sh\n{}", source)
+      };
+      let tempdir = TempDir::new().expect("fixme");
+      let script_file = tempdir.path().join("script");
+      tokio::fs::write(&script_file, source).await.expect("fixme");
+      run!(%"chmod +x", &script_file);
+      run!(%"cat", &script_file);
+      let output = tokio::process::Command::new(script_file)
```

Comment:
Not necessary to do now, but we should remember to check the status code of the script. Not sure what to do if it fails.